lib/features/push_notification/presentation/websocket/web_socket_queue_handler.dart

```diff
@@ -0,0 +1,107 @@
+
+import 'dart:async';
+import 'dart:collection';
+
+import 'package:core/utils/app_logger.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:tmail_ui_user/features/push_notification/presentation/websocket/web_socket_message.dart';
+
+typedef ProcessMessageCallback = Future<void> Function(WebSocketMessage message);
+typedef OnErrorCallback = void Function(dynamic error, StackTrace stackTrace);
+
+class WebSocketQueueHandler {
+  final Queue<WebSocketMessage> _messageQueue = Queue<WebSocketMessage>();
+  final Set<String> _processedMessageIds = {};
```

Comment:
we should have fixed size for queue and _processedMessageIds to prevent memory leak.

Reply:
done